gensim/models/ensemblelda.py

```diff
@@ -0,0 +1,1280 @@
+                "`corpus` keyword argument.")
+
+        if type(topic_model_class) == type and issubclass(topic_model_class, ldamodel.LdaModel):
+            self.topic_model_class = topic_model_class
+        else:
+            kinds = {
+                "lda": ldamodel.LdaModel,
+                "ldamulticore": ldamulticore.LdaMulticore
+            }
+            if topic_model_class not in kinds:
+                raise ValueError(
+                    "topic_model_class should be one of 'lda', 'ldamulticode' or a model inheriting from LdaModel")
+            self.topic_model_class = kinds[topic_model_class]
+
+        self.num_models = num_models
+        self.gensim_kw_args = gensim_kw_args
+
+        self.memory_friendly_ttda = memory_friendly_ttda
+
+        self.distance_workers = distance_workers
+        self.masking_threshold = masking_threshold
+        self.masking_method = masking_method
+
+        # this will provide the gensim api to the ensemble basically
+        self.classic_model_representation = None
+
+        # the ensembles state
+        self.random_state = utils.get_random_state(random_state)
+        self.sstats_sum = 0
+        self.eta = None
+        self.tms = []
+        # initialize empty topic term distribution array
+        self.ttda = np.empty((0, len(gensim_kw_args["id2word"])))
+        self.asymmetric_distance_matrix_outdated = True
+
+        # in case the model will not train due to some
+        # parameters, stop here and don't train.
+        if num_models <= 0:
+            return
+        if gensim_kw_args.get("corpus") is None:
+            return
+        if "iterations" in gensim_kw_args and gensim_kw_args["iterations"] <= 0:
+            return
+        if "passes" in gensim_kw_args and gensim_kw_args["passes"] <= 0:
+            return
+
+        logger.info("generating {} topic models...".format(num_models))
+
+        if ensemble_workers > 1:
+            self._generate_topic_models_multiproc(num_models, ensemble_workers)
+        else:
+            # singlecore
+            self._generate_topic_models(num_models)
+
+        self._generate_asymmetric_distance_matrix()
+        self._generate_topic_clusters(epsilon, min_samples)
+        self._generate_stable_topics(min_cores)
+
+        # create model that can provide the usual gensim api to the stable topics from the ensemble
+        self.generate_gensim_representation()
+
+    def get_topic_model_class(self):
+        """Get the class that is used for :meth:`gensim.models.EnsembleLda.generate_gensim_representation`."""
+        if self.topic_model_class is None:
+            try:
+                module = importlib.import_module(self.topic_model_module_string)
+                self.topic_model_class = getattr(module, self.topic_model_class_string)
+                del self.topic_model_module_string
+                del self.topic_model_class_string
+            except ModuleNotFoundError:
```

Comment:
thanks. Can you explain why .format should not be used? Since python2 support was dropped it should be compatible with everybody who runs gensim, shouldn't it?

Reply:
the error messages are actually a tiny bit different